File: src/main/java/org/pixeltime/enchantmentsenhance/events/enchantments/Plow.java
package org.pixeltime.enchantmentsenhance.events.enchantments;

import com.sk89q.worldguard.bukkit.WGBukkit;
import org.bukkit.ChatColor;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerInteractEvent;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Plow implements Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    public void onInteract(final PlayerInteractEvent playerInteractEvent) {
        if (playerInteractEvent.isCancelled()) {
            return;
        }
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "plow"));
        final Player player = playerInteractEvent.getPlayer();
        if (player.getItemInHand() != null && player.getItemInHand().hasItemMeta() && player.getItemInHand().getItemMeta().hasLore()) {
            final Block clickedBlock = playerInteractEvent.getClickedBlock();
            if (clickedBlock.getType() != Material.DIRT && clickedBlock.getType() != Material.GRASS) {
                return;
            }
            if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                final double n = 0.5;
                final Location location = clickedBlock.getLocation();
                for (double n2 = location.getBlockX() - n; n2 <= location.getBlockX() + n; ++n2) {
                    for (double n3 = location.getBlockZ() - n; n3 <= location.getBlockZ() + n; ++n3) {
                        final Block block = location.getWorld().getBlockAt(new Location(clickedBlock.getWorld(), n2, (double) clickedBlock.getY(), n3));
                        if (block.getType() != Material.GRASS && block.getType() != Material.DIRT) {
                            return;
                        }
                        if (SettingsManager.enchant.getBoolean("allow-worldguard") && !WGBukkit.getPlugin().canBuild(player, block)) {
                            return;
                        }
                        block.setType(Material.SOIL);
                    }
                }
            }
            if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " II")) {
                final double n4 = 1.0;
                final Location location2 = clickedBlock.getLocation();
                for (double n5 = location2.getBlockX() - n4; n5 <= location2.getBlockX() + n4; ++n5) {
                    for (double n6 = location2.getBlockZ() - n4; n6 <= location2.getBlockZ() + n4; ++n6) {
                        final Block block2 = location2.getWorld().getBlockAt(new Location(clickedBlock.getWorld(), n5, (double) clickedBlock.getY(), n6));
                        if (block2.getType() != Material.GRASS && block2.getType() != Material.DIRT) {
                            return;
                        }
                        if (SettingsManager.enchant.getBoolean("allow-worldguard") && !WGBukkit.getPlugin().canBuild(player, block2)) {
                            return;
                        }
                        block2.setType(Material.SOIL);
                    }
                }
            }
            if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " III")) {
                final double n7 = 1.5;
                final Location location3 = clickedBlock.getLocation();
                for (double n8 = location3.getBlockX() - n7; n8 <= location3.getBlockX() + n7; ++n8) {
                    for (double n9 = location3.getBlockZ() - n7; n9 <= location3.getBlockZ() + n7; ++n9) {
                        final Block block3 = location3.getWorld().getBlockAt(new Location(clickedBlock.getWorld(), n8, (double) clickedBlock.getY(), n9));
                        if (block3.getType() != Material.GRASS && block3.getType() != Material.DIRT) {
                            return;
                        }
                        if (SettingsManager.enchant.getBoolean("allow-worldguard") && !WGBukkit.getPlugin().canBuild(player, block3)) {
                            return;
                        }
                        block3.setType(Material.SOIL);
                    }
                }
            }
            if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " IV")) {
                final double n10 = 2.0;
                final Location location4 = clickedBlock.getLocation();
                for (double n11 = location4.getBlockX() - n10; n11 <= location4.getBlockX() + n10; ++n11) {
                    for (double n12 = location4.getBlockZ() - n10; n12 <= location4.getBlockZ() + n10; ++n12) {
                        final Block block4 = location4.getWorld().getBlockAt(new Location(clickedBlock.getWorld(), n11, (double) clickedBlock.getY(), n12));
                        if (block4.getType() != Material.GRASS && block4.getType() != Material.DIRT) {
                            return;
                        }
                        if (SettingsManager.enchant.getBoolean("allow-worldguard") && !WGBukkit.getPlugin().canBuild(player, block4)) {
                            return;
                        }
                        block4.setType(Material.SOIL);
                    }
                }
            }
            if (player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " V")) {
                final double n13 = 2.5;
                final Location location5 = clickedBlock.getLocation();
                for (double n14 = location5.getBlockX() - n13; n14 <= location5.getBlockX() + n13; ++n14) {
                    for (double n15 = location5.getBlockZ() - n13; n15 <= location5.getBlockZ() + n13; ++n15) {
                        final Block block5 = location5.getWorld().getBlockAt(new Location(clickedBlock.getWorld(), n14, (double) clickedBlock.getY(), n15));
                        if (block5.getType() != Material.GRASS && block5.getType() != Material.DIRT) {
                            return;
                        }
                        if (SettingsManager.enchant.getBoolean("allow-worldguard") && !WGBukkit.getPlugin().canBuild(player, block5)) {
                            return;
                        }
                        block5.setType(Material.SOIL);
                    }
                }
            }
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantment/Plow.kt
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.event.enchantment

import com.sk89q.worldguard.bukkit.WGBukkit
import org.bukkit.ChatColor
import org.bukkit.Location
import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerInteractEvent
import org.pixeltime.enchantmentsenhance.manager.IM
import org.pixeltime.enchantmentsenhance.manager.KM
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Plow : Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    fun onInteract(playerInteractEvent: PlayerInteractEvent) {
        if (playerInteractEvent.isCancelled) {
            return
        }
        val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "plow"))
        val player = playerInteractEvent.player
        val armorContents = IM.getItemList(player)
        for (itemStack in armorContents) {

            val level = KM.getLevel(translateAlternateColorCodes, itemStack.itemMeta.lore)
            if (level > 0) {
                val clickedBlock = playerInteractEvent.clickedBlock
                if (clickedBlock.type != Material.DIRT && clickedBlock.type != Material.GRASS) {
                    return
                }
                val n = 0.5
                val location = clickedBlock.location
                var n2 = location.blockX - n
                while (n2 <= location.blockX + n) {
                    var n3 = location.blockZ - n
                    while (n3 <= location.blockZ + n) {
                        val block = location.world.getBlockAt(Location(clickedBlock.world, n2, clickedBlock.y.toDouble(), n3))
                        if (block.type != Material.GRASS && block.type != Material.DIRT) {
                            return
                        }
                        if (SettingsManager.enchant.getBoolean("allow-worldguard") && !WGBukkit.getPlugin().canBuild(player, block)) {
                            return
                        }
                        block.type = Material.SOIL
                        ++n3
                    }
                    ++n2
                }

            }
        }
    }
}